Update code-conventions.md (#536)
demisto/content-docs: docs/integrations/code-conventions.md

```
**Note:** By default, ignore_auto_extract is set to ```False```.

### Commands in integration with paging

> When dealing with a command in an integration, including paging mechanism (```page``` and ```page_size``` like arguments) convention is:
> - Add to the command ```page```, ```page_size``` and ```limit``` arguments. Where limit argument indicates how many results/incidents/indicators to retrieve from server starting from offset 0, making the number of needed API calls to achieve that limit, and ```page``` and ```page_size``` will be used the normal paging mechanism of the integration.

Reviewer: ```suggestion
When working on a command that supports pagination (usually has API parameters like`page` and/or `page size`) with a maximal page size enforced by the API, our best practice is to create a command to support two different use-cases with these 3 corresponding integer arguments:
1. `page` 
2. `page_size` 
3. `limit` 

**Manual Pagination:** The user wants to control the pagination on its own by using the `page` and `page size` arguments. To achieve this, the command will simply pass the `page` and `page size` values on to the API request.

**Automatic Pagination:** The user does not want to work with pages only with a number of results argument. In this case, the `limit` argument will be used to aggregate results by iterating over the necessary pages from the first page until collecting all the needed results. This implies a pagination loop mechanism will be implemented behind the scenes. For example, if the limit value received is 250 and the maximal page size enforced by the API is 100, the command will need to perform 3 API calls (pages 1,2, and 3) to collect the 250 requested results.
```

### Credentials

> When dealing with integrations which require user credentials (such as username/password, API token, etc..) the expected way is to use credentials parameter.

Reviewer: ```suggestion
When working on integrations that require user credentials (such as username/password, API token/key, etc..) the best practice is to use the `credentials` parameter type.
```

> example for parameter configuration for integration using username and password:

Reviewer: ```suggestion
Example when using username and password:
```

Reviewer: Add a screenshot of the integration config here

> - display: Username
>   name: credentials
>   type: 9
>   required: true

Reviewer: (Note that the `\` before the backticks are not needed here it is just due to Github markdown parsing)

```suggestion
In the YML file:
\```
- display: Username
  name: credentials
  type: 9
  required: true
\```

```

> example for parameter configuration for integration using API token:

Reviewer: ```suggestion
Example when using an API Token/Key:
```

Reviewer: Add a screenshot of the integration config here too

> - displaypassword: API Token
>   name: credentials
>   type: 9
>   required: true
>   hiddenusername: true

Reviewer: ```suggestion
\```
- displaypassword: API Token
  name: credentials
  type: 9
  required: true
  hiddenusername: true
\```
```

> Using parameter type of credentials is needed even for case of API token and not username/password as it provides the user the flexibility of 'switching to credentials' where he could input his credentials instead of his access information.

Reviewer: ```suggestion
Using credentials parameter type is always recommended (even when working with API token\key) as it provides the user the flexibility of using the [XSOAR credentials vault](https://xsoar.pan.dev/docs/reference/articles/managing-credentials) feature when configuring the integration for the first time.
```

```
**Note:** By default, ignore_auto_extract is set to ```False```.

### Pagination in integration commands
When working on a command that supports pagination (usually has API parameters like `page` and/or `page size`) with a maximal page size enforced by the API, our best practice is to create a command that will support two different use-cases with the following 3 integer arguments:
1. `page` 
2. `page_size` 
3. `limit` 

> **The two use cases"** 

Reviewer: ```suggestion
**The two use cases** 
```

> **Manual Pagination:** The user wants to control the pagination on its own by using the `page` and `page size` arguments. To achieve this, the command will simply pass the `page` and `page size` values on to the API request.
> 
> **Automatic Pagination:** The user does not want to work with pages only with a number of results argument. In this case, the `limit` argument will be used to aggregate results by iterating over the necessary pages from the first page until collecting all the needed results. This implies a pagination loop mechanism will be implemented behind the scenes. For example, if the limit value received is 250 and the maximal page size enforced by the API is 100, the command will need to perform 3 API calls (pages 1,2, and 3) to collect the 250 requested results.

Reviewer: ```suggestion
**The two use cases"** 
- **Manual Pagination:** The user wants to control the pagination on its own by using the `page` and `page size` arguments. To achieve this, the command will simply pass the `page` and `page size` values on to the API request.
- **Automatic Pagination:** The user does not want to work with pages, but only with a number of total results. In this case, the `limit` argument will be used to aggregate results by iterating over the necessary pages from the first page until collecting all the needed results. This implies a pagination loop mechanism will be implemented behind the scenes. For example, if the limit value received is 250 and the maximal page size enforced by the API is 100, the command will need to perform 3 API calls (pages 1,2, and 3) to collect the 250 requested results.
```

### Credentials
When working on integrations that require user credentials (such as username/password, API token/key, etc..) the best practice is to use the `credentials` parameter type.
Example when using username and password:
In the YML file:
```yml
- display: Username
  name: credentials
  type: 9
  required: true
```
In Demisto UI:
<img width="758" src="../doc_imgs/integrations/credentials_username_password.png"></img>

Example when using an API Token/Key:
```yml
- displaypassword: API Token
  name: credentials
  type: 9
  required: false
  hiddenusername: true
```
In Demisto UI:
<img width="758" src="../doc_imgs/integrations/credentials_api_token.png"></img>
  

> Using credentials parameter type is always recommended (even when working with API token\key) as it provides the user the flexibility of using the [XSOAR credentials vault](https://xsoar.pan.dev/docs/reference/articles/managing-credentials) feature when configuring the integration for the first time.

Reviewer: **Using username and password:**

- **In Demisto UI:**
![image](../doc_imgs/integrations/credentials_username_password.png)

- **In the YML file:**
```yml
- display: Username
  name: credentials
  type: 9
  required: true
```

**Using an API Token/Key:**

- **In Demisto UI:**
![image](../doc_imgs/integrations/credentials_api_token.png)

- **In the YML file:**
```yml
- displaypassword: API Token
  name: credentials
  type: 9
  required: false
  hiddenusername: true
```
  
Using credentials parameter type is always recommended (even when working with API token\key) as it provides the user the flexibility of using the [XSOAR credentials vault](../docs/reference/articles/managing-credentials) feature when configuring the integration for the first time.
